File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/Tva.Phasors/HeaderFrameParsingState.vb
'*******************************************************************************************************
'  HeaderFrameParsingState.vb - Header frame parsing state class
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: James R Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  01/14/2005 - James R Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

' This class represents the protocol independent common implementation the parsing state of a header frame that can be sent or received from a PMU.
<CLSCompliant(False)> _
Public Class HeaderFrameParsingState

    Inherits ChannelFrameParsingStateBase(Of IHeaderCell)
    Implements IHeaderFrameParsingState

    Public Sub New(ByVal cells As HeaderCellCollection, ByVal frameLength As Int16, ByVal dataLength As Int16)

        MyBase.New(cells, frameLength, AddressOf HeaderCell.CreateNewHeaderCell)

        CellCount = dataLength

    End Sub

    Public Overrides ReadOnly Property InheritedType() As System.Type
        Get
            Return Me.GetType()
        End Get
    End Property

    Public Overridable Shadows ReadOnly Property Cells() As HeaderCellCollection Implements IHeaderFrameParsingState.Cells
        Get
            Return MyBase.Cells
        End Get
    End Property

End Class

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/HeaderFrameParsingState.cs
//*******************************************************************************************************
//  HeaderFrameParsingState.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  01/14/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace PhasorProtocols
{
    /// <summary>
    /// Represents the protocol independent common implementation of the parsing state used by any <see cref="IHeaderFrame"/>.
    /// </summary>
    public class HeaderFrameParsingState : ChannelFrameParsingStateBase<IHeaderCell>, IHeaderFrameParsingState
    {
        #region [ Constructors ]

        /// <summary>
        /// Creates a new <see cref="HeaderFrameParsingState"/> from specified parameters.
        /// </summary>
        /// <param name="parsedBinaryLength">Binary length of the <see cref="IHeaderFrame"/> being parsed.</param>
        /// <param name="dataLength">Length of data in <see cref="IHeaderFrame"/> being parsed (i.e., number of cells).</param>
        public HeaderFrameParsingState(int parsedBinaryLength, int dataLength)
            : base(parsedBinaryLength, HeaderCell.CreateNewCell)
        {
            CellCount = dataLength;
        }

        #endregion
    }
}